cxx/clean_relation.hh

```diff
@@ -40,7 +41,9 @@ class CleanRelation : public Relation<T> {
   // list of domain pointers
   const std::vector<Domain*> domains;
   // Distribution or Emission spec over the relation's codomain.
-  const std::variant<DistributionSpec, EmissionSpec> prior_spec;
+  const std::string prior_spec;
+  // Is the codomain an emission?
+  const bool codomain_is_emission;
```

Comment:
I'm open to replacing codomain_is_emission with something else.  Here are some possibilities:

1) We resurrect DistributionSpec and EmissionSpec as just wrappers around a string.  (This is clearly the same complexity as before.)

2) If we think we will eventually need or want run time dynamism, we could allow get_emission and get_distribution to return nullptrs, which would let us run the string prior_spec through both of them to see which one succeeds.

3) We replace CleanRelation's constructor with two static make methods:  make_clean_relation_from_distribution and make_clean_relation_from_emission.  This is basically #1 with different syntax.

4) We make CleanRelation into a non-concrete base class, and give it two children:  CleanDistributionRelation and CleanEmissionRelation.  Almost all of the code except make_new_distribution still lives in CleanRelation.

Reply:
I like `1`. (For `2`, it doesn't strike me that we'd need/want runtime dynamism, and for 3/4, IMO we don't want to be treating distributions and emissions separately in CleanRelation -- that adds unnecessary complexity, since CleanRelation can/does happily cast Emissions to Distributions when using them as clusters).

If we use `1`, then DistributionSpec and EmissionSpec are storing strings that are used as enums, which again, IMO, should be enums.

Reply:
I changed the code to use #1.